EDU-15945: Update FastStore SEO (#2140)
vtexdocs/dev-portal-content: docs/faststore/docs/seo/optimizing-seo-by-managing-lazy-loading-on-plp-and-pdp.mdx

This approach reduces initial page load time and improves the browsing experience, especially on slower devices and networks. However, since sections outside the viewport are not included in the initial HTML, there can be SEO implications, such as search engines not indexing them immediately.

This guide explains how to manage lazy loading for custom sections to balance SEO and performance.

> ❗ Disabling lazy loading will most likely increase initial page load time. Apply this practice only to critical, above-the-fold content or sections with SEO-relevant content that must render on the initial page load. To help you decide, see [Key concepts before implementation](#key-concepts-before-implementation) and [Deciding when to lazy load](#step-1--deciding-when-to-lazy-load).

## Key concepts before implementation

`LazyLoadingSection` and `ViewportObserver` work together in FastStore to improve initial page load speed.

* `LazyLoadingSection`: A React wrapper that delays the rendering of content (e.g., blocks) and hydration of non-critical blocks until the content and non-critical blocks approach the viewport.  

> * `ViewportObserver`: A small utility that detects when content is approaching the viewport, triggering the `LazyLoadingSection` to render the content.

Reviewer: ```suggestion
* `ViewportObserver`: A utility that detects when content is approaching the viewport and triggers the `LazyLoadingSection` to render the content.
```

### Server-Side Rendering (SSR) and lazy loading

Server-side rendering provides crucial HTML in the initial document, allowing users and crawlers to access content without executing JavaScript. **Sections wrapped with lazy loading are not included in this initial HTML.** They are only rendered and hydrated when the sections are near the viewport.

When creating custom sections, you can add the **Skip lazy loading** flag to a section to disable lazy loading for that section only in FastStore's PDP and PLP templates. With lazy loading disabled, bots can access all content immediately without needing to scroll or run scripts, which may speed up the crawling process.

However, they still need to download and process all media files, which can quickly consume crawl budget, slow down the indexing of other pages, and reduce the number of pages indexed. Larger initial payloads can also negatively impact user experience and make the site seem slower to bots.

> 

Reviewer: ```suggestion
```

title: "Optimizing SEO by managing lazy loading on PLPs and PDPs"
---

Lazy loading is a method that delays the loading of non-essential resources, such as images or sections of a page, until they are actually needed. FastStore uses lazy loading for all native store sections that are out of the viewport by default. Each section is wrapped in a component that checks if it is within the viewport. If it is not, it is loaded lazily.

This approach reduces initial page load time and improves the browsing experience, especially on slower devices and networks. However, since sections outside the viewport are not included in the initial HTML, there can be SEO implications, such as search engines not indexing them immediately.

This guide explains how to manage lazy loading for custom sections to balance SEO and performance.

> > ❗ Disabling lazy loading will most likely increase initial page load time. Apply this practice only to critical, above-the-fold content. To help you decide, see [How lazy loading works in FastStore](#how-lazy-loading-works-in-faststore) and [Deciding when to lazy load](#deciding-when-to-lazy-load).

Reviewer 2: Here I'd highlight that another scenario for disabling lazy loading is for sections with SEO-relevant content that needs to be rendered in the first page load.

> > ❗ Disabling lazy loading will most likely increase initial page load time. Apply this practice only to critical, above-the-fold content. To help you decide, see [How lazy loading works in FastStore](#how-lazy-loading-works-in-faststore) and [Deciding when to lazy load](#deciding-when-to-lazy-load).

Reviewer 3: ```suggestion
> ❗ Disabling lazy loading will most likely increase initial page load time. Apply this practice only to critical, above-the-fold content or to sections with SEO‑relevant content that must render on the initial page load. To help you decide, see [How lazy loading works in FastStore](#how-lazy-loading-works-in-faststore) and [Deciding when to lazy load](#deciding-when-to-lazy-load).
```

| **To balance server load during high traffic** | It reduces server spikes by loading only the images visible on the screen, keeping the browsing experience fast. | Category pages like "Dresses" or "Jeans" may have hundreds of product image previews. With lazy loading, only the images visible above the fold load immediately, which reduces server spikes and keeps the browsing experience fast. |

### When to skip lazy loading ❌ 

| Context | Reason to avoid | Example |
| :---- | :---- | :---- |
| **There is a risk of hiding critical page elements from crawlers** | Search engines might not see all content if it is lazy-loaded, potentially harming your site's SEO. | A "New Arrivals" grid on your homepage displays featured items from several categories. Lazy loading is skipped here to ensure search engines can index all new products shown in this grid. |

> | **Do not disable lazy loading globally** | Apply lazy loading selectively. Rendering everything upfront increases payloads, exhausts crawl budget, and slows indexing. | A homepage with numerous campaign images should lazy load secondary media while keeping critical content immediately available. |

Reviewer 2: Maybe this part can be a bit confusing. Because in the column "Context" we have "Do not disable lazy loading globally". And we have the column "Reasons to avoid". So maybe it can be like a double negative. Maybe the "Context" value for this line should be "Disable Lazy loading globally". What do you think?

## Instructions

### Step 1: Deciding when to lazy load

Use lazy loading strategically to prioritize the most important content for crawlers. If essential content is not present in the initial HTML, search engines may consume more crawl budget to execute \`js\` scripts, such as product grids on a PLP.

> ### When to implement lazy loading ✅

Reviewer: ```suggestion
#### When to use lazy loading 
```

| **To improve page speed for a media-heavy page** | Optimizes resource usage and ensures the most important content is prioritized and loads first. | A PDP includes a "Complete the Look" carousel at the bottom. Since this is a supplementary suggestion, lazy loading ensures the main product details are prioritized and the carousel images load only when the user scrolls to that section. |
| **The content doesn't affect SEO or crawling** | It will not negatively impact search engine visibility since the content isn't critical for SEO or structured data. | A store's homepage has a "Fashion Trends" slider in the footer with inspirational videos. Since this content does not impact structured data for SEO, lazy loading it will not negatively affect search engine visibility. |
| **To balance server load during high traffic** | It reduces server spikes by loading only the images visible on the screen, keeping the browsing experience fast. | Category pages like "Dresses" or "Jeans" may have hundreds of product image previews. With lazy loading, only the images visible above the fold load immediately, which reduces server spikes and keeps the browsing experience fast. |

> ### When to skip lazy loading ❌ 

Reviewer: ```suggestion
#### When not to use lazy loading 
```
